### ISSUE TITLE
Unkown error after refresh on accepted share page

### ISSUE THREAD

**Issue by MorrisJobke** [owncloud/core#13696]
* Share a folder on server1 with mjoob at server2
* login as mjob on server2 and accept the share
* refresh

## expected behaviour

* share is listed

## actual behaviour

* empty filelist and "unknown error"
* after a second refresh everything is fine. It's just the first refresh

![s2s-unknown-error](https://cloud.githubusercontent.com/assets/245432/5910064/9992a960-a5b6-11e4-8c15-97d37ba2b0ee.png)

Both servers are on current master bd88874

cc @PVince81 @schiesbn

**Comment by MorrisJobke** [comment]
```
{"reqId":"6f5e747397ee98c1500e47dc1c1ce8ad","remoteAddr":"127.0.0.1","app":"files","message":"Exception: {\"Message\":\"An exception occurred while executing 'INSERT INTO \\\"oc_storages\\\" (\\\"id\\\") VALUES(?)' with params [\\\"shared::e84560e76cea7251dd205a93c205c116\\\"]:\\n\\nSQLSTATE[23000]: Integrity constraint violation: 19 UNIQUE constraint failed: oc_storages.id\",\"Code\":0,\"Trace\":\"#0 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/DBALException.php(116): Doctrine\\\\DBAL\\\\Driver\\\\AbstractSQLiteDriver->convertException('An exception oc...', Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException))\\n#1 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Statement.php(172): Doctrine\\\\DBAL\\\\DBALException::driverExceptionDuringQuery(Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOSqlite\\\\Driver), Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException), 'INSERT INTO \\\"oc...', Array)\\n#2 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/private\\\/db\\\/statementwrapper.php(63): Doctrine\\\\DBAL\\\\Statement->execute(Array)\\n#3 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/private\\\/db.php(141): OC_DB_StatementWrapper->execute(Array)\\n#4 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/private\\\/files\\\/cache\\\/storage.php(39): OC_DB::executeAudited('INSERT INTO `*P...', Array)\\n#5 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/private\\\/files\\\/cache\\\/cache.php(53): OC\\\\Files\\\\Cache\\\\Storage->__construct(Object(OCA\\\\Files_Sharing\\\\External\\\\Storage))\\n#6 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/apps\\\/files_sharing\\\/lib\\\/external\\\/cache.php(26): OC\\\\Files\\\\Cache\\\\Cache->__construct(Object(OCA\\\\Files_Sharing\\\\External\\\\Storage))\\n#7 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/apps\\\/files_sharing\\\/lib\\\/external\\\/storage.php(107): OCA\\\\Files_Sharing\\\\External\\\\Cache->__construct(Object(OCA\\\\Files_Sharing\\\\External\\\\Storage), 'http:\\\/\\\/localhos...', 'mjob')\\n#8 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/wrapper.php(371): OCA\\\\Files_Sharing\\\\External\\\\Storage->getCache('', Object(OCA\\\\Files_Trashbin\\\\Storage))\\n#9 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/wrapper.php(371): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->getCache('', Object(OCA\\\\Files_Trashbin\\\\Storage))\\n#10 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/private\\\/files\\\/view.php(954): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->getCache('')\\n#11 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/private\\\/files\\\/filesystem.php(782): OC\\\\Files\\\\View->getFileInfo('\\\/', true)\\n#12 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/apps\\\/files\\\/ajax\\\/list.php(12): OC\\\\Files\\\\Filesystem::getFileInfo('\\\/')\\n#13 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/private\\\/route\\\/route.php(135) : runtime-created function(1): require_once('\\\/home\\\/mjob\\\/Proj...')\\n#14 [internal function]: __lambda_func(Array)\\n#15 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/private\\\/route\\\/router.php(250): call_user_func('\\\\x00lambda_1154', Array)\\n#16 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/base.php(763): OC\\\\Route\\\\Router->match('\\\/apps\\\/files\\\/aja...')\\n#17 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/index.php(28): OC::handleRequest()\\n#18 {main}\",\"File\":\"\\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Driver\\\/AbstractSQLiteDriver.php\",\"Line\":47}","level":4,"time":"2015-01-26T23:01:03+00:00","method":"GET","url":"\/master\/index.php\/apps\/files\/ajax\/list.php?dir=%2F&sort=name&sortdirection=asc"}
```

This error is also gone with #13691

**Comment by PVince81** [comment]
Ah, "integrity constrain violation" on oc_storages... maybe the scanner is running concurrently and trying to insert that entry twice at the same time. CC @icewind1991 

We should revamp the scanner and find a way to avoid having two scanner instances scanning the same folders/storages at the same time: https://github.com/owncloud/core/issues/13703

**Comment by MorrisJobke** [comment]
@PVince81 Is this case fully covered by #13703? Then please close this ticket, because it is now fixed with #13691 ;)

**Comment by PVince81** [comment]
Hmm ok, then let's follow up in https://github.com/owncloud/core/issues/13703 for a more general fix
